lib/Mojo/WebSocketProxy/Dispatcher/Parser.pm

```diff
@@ -0,0 +1,104 @@
+package Mojo::WebSocketProxy::Dispatcher::Parser;
+
+use strict;
+use warnings;
+
+use Mojo::WebSocketProxy::Dispatcher::Config;
+
+sub parse_req {
+    my ($c, $req_storage) = @_;
+
+    my $result;
+    my $args = $req_storage->{args};
+    if (ref $args ne 'HASH') {
+        # for invalid call, eg: not json
+        $req_storage->{args} = {};
+        $result = $c->wsp_error('error', 'BadRequest', 'The application sent an invalid request.');
+    }
+
+    $result = _check_sanity($c, $req_storage) unless $result;
+
+    return $result;
+}
+
+sub _check_sanity {
```

Comment:
No. The question is - why we do sanity check for JSON at all? How JSON can be insane, and what errors might that introduce?

Now we validate json, the it has alpha-numeric keys and alpha-numeric-values with some subset of unicode symbols. The question is  - what is the sense of that validation at all?